app.py

```diff
@@ -8,13 +8,55 @@
     PicklePersistence,
 )
 from ptbcontrib.ptb_jobstores.sqlalchemy import PTBSQLAlchemyJobStore
+
+import grpc
+from opentelemetry import metrics
+from opentelemetry.exporter.otlp.proto.grpc.metric_exporter import (
+    OTLPMetricExporter,
+)
+from opentelemetry.sdk import metrics as sdkmetrics
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import (
+    AggregationTemporality,
+    PeriodicExportingMetricReader,
+)
+from opentelemetry.sdk.resources import Resource
+
 from src.custom_filters import filter_bot_added
 from src.logging import tg_logger
 from src import handlers
 import os
+import time
+
+
+temporality_delta = {
+    sdkmetrics.Counter: AggregationTemporality.DELTA,
+    sdkmetrics.UpDownCounter: AggregationTemporality.DELTA,
+    sdkmetrics.Histogram: AggregationTemporality.DELTA,
+    sdkmetrics.ObservableCounter: AggregationTemporality.DELTA,
+    sdkmetrics.ObservableUpDownCounter: AggregationTemporality.DELTA,
+    sdkmetrics.ObservableGauge: AggregationTemporality.DELTA,
+}
 
 
 def main():
+    dsn = os.environ["UPTRACE_DSN"]
+
+    exporter = OTLPMetricExporter(
+        endpoint="otlp.uptrace.dev:4317",
+        headers=(("uptrace-dsn", dsn),),
+        timeout=5,
+        compression=grpc.Compression.Gzip,
+        preferred_temporality=temporality_delta,
+    )
+    reader = PeriodicExportingMetricReader(exporter)
+
+    resource = Resource(
+        attributes={"service.name": "wachter", "service.version": "2.0.0"}
```

Comment:
you can also add `DEPLOYMENT_ENVIRONMENT` attribute to separate prod/testing, see `src/logging.py` on how to use that